pennylane/tape/qnode.py

```diff
@@ -576,7 +611,16 @@ def to_torch(self, dtype=None):
             import torch
             from pennylane.tape.interfaces.torch import TorchInterface
 
-            self.interface = "torch"
+            if self.interface != "torch" and self.interface is not None:
+                # Since the interface is changing, need to re-validate the tape class.
+                self._tape, interface, diff_method, self.device = self.get_tape(
+                    self._original_device, "torch", self.diff_method
+                )

```

Comment:
Ohh wait, I think I get these parts now! The idea is if we change the interface, it'll automatically check if it can change the device under the hood to ideally maintain backprop support?

Do we need to do this for torch?

Reply:
Yep, because this allows us to go _backward_ 🙂 

E.g., a QNode running on `default.qubit.tf` and backprop will be able to be converted to a QNode running on default.qubit with parameter-shift!

Even better, if the user explicitly asked for backprop, the QNode remembers that preference (since we save it as `self.diff_method`), and calling `qnode.to_torch()` will raise an error since backprop is not supported. If `self.diff_method` was `best`, it will work with no issue.